# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 4cda148ef28f
# Full Hash: 4cda148ef28f72a0ca30b68919332996e667eec6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Create a separate WebGLContext::PresentInto function for WebXR. r=jgilbert
#   
#   - Do not apply mIsCanvasDirty bailout when presenting from a opaque FBO
#   back buffer.
#   - Create a separate WebGLContext::PresentInto with more specific XR
# ==============================================================================

diff -r 0b191b707750 -r 4cda148ef28f dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Mon Jun 15 18:25:58 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Mon Jun 15 18:26:00 2020 +0000
@@ -101,7 +101,7 @@
 namespace layers {
 class CompositableHost;
 class SurfaceDescriptor;
-}
+}  // namespace layers
 
 namespace webgl {
 class AvailabilityRunnable;
@@ -483,6 +483,7 @@
   // Present to compositor
  private:
   bool PresentInto(gl::SwapChain& swapChain);
+  bool PresentIntoXR(gl::SwapChain& swapChain, const gl::MozFramebuffer& fb);
 
  public:
   void Present(WebGLFramebuffer*, layers::TextureType);
